lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ConnectorContextModule.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base;
+
+import com.google.inject.Module;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Tracer;
+import io.trino.spi.Node;
+import io.trino.spi.NodeManager;
+import io.trino.spi.NodeVersion;
+import io.trino.spi.PageIndexerFactory;
+import io.trino.spi.PageSorter;
+import io.trino.spi.VersionEmbedder;
+import io.trino.spi.connector.ConnectorContext;
+import io.trino.spi.connector.MetadataProvider;
+import io.trino.spi.type.TypeManager;
+
+import static io.airlift.configuration.ConfigurationAwareModule.combine;
+
+public final class ConnectorContextModule
+{
+    private ConnectorContextModule() {}
+
+    public static Module from(String catalogName, ConnectorContext context)
+    {
+        return combine(
+                ConnectorContextModule.from(context),
+                new CatalogNameModule(catalogName));
+    }
+
+    public static Module from(ConnectorContext context)
```

Comment:
can we instead of a lambda here have `ConnectorContextModule` implement module and have a configure method?